test/ca/mcgill/cs/stg/jetuml/commands/TestComponudCommand.java

```diff
@@ -0,0 +1,272 @@
+    
+    @Before
+    public void setUp() throws Exception
+    {
+        aGraph = new ClassDiagramGraph();
+        aClassNode = new ClassNode();
+        aActorNode = new ActorNode();
+        aCallEdge = new CallEdge();
+        
+        aAddClassNodeCommand = new AddNodeCommand(aGraph, aClassNode);
+        aAddEdgeCommand = new AddEdgeCommand(aGraph, aCallEdge);
+        aAddActorNodeCommand = new AddNodeCommand(aGraph, aActorNode);
+        aMoveCommand1 = new MoveCommand(aGraph, aClassNode, 5, 5);
+        aMoveCommand2 = new MoveCommand(aGraph, aClassNode, -3, 3);
+        aDeletNodeCommand = new DeleteNodeCommand(aGraph, aClassNode);
+        
+        aNodesToBeRemoved = aGraph.getClass().getSuperclass().getDeclaredField("aNodesToBeRemoved");
+        aNodesToBeRemoved.setAccessible(true);
+        aEdgesToBeRemoved = aGraph.getClass().getSuperclass().getDeclaredField("aEdgesToBeRemoved");
+        aEdgesToBeRemoved.setAccessible(true);
+        aNeedsLayout = aGraph.getClass().getSuperclass().getDeclaredField("aNeedsLayout");
+        aNeedsLayout.setAccessible(true);
+        aRootNodes = aGraph.getClass().getSuperclass().getDeclaredField("aRootNodes");
+        aRootNodes.setAccessible(true);
+        
+        aCompoundCommand1 = new CompoundCommand();
+        aCompoundCommand1.add(aAddClassNodeCommand);
+        aCompoundCommand1.add(aAddEdgeCommand);
+        aCompoundCommand1.add(aMoveCommand1);
+        
+        aCompoundCommand2 = new CompoundCommand();
+        aCompoundCommand2.add(aAddActorNodeCommand);
+        aCompoundCommand2.add(aMoveCommand2);
+        
+        aCompoundCommand3 = new CompoundCommand();
+        aCompoundCommand3.add(aCompoundCommand1);
+        aCompoundCommand3.add(aCompoundCommand2);
+        
+        aCompoundCommand4 = new CompoundCommand();
+        aCompoundCommand4.add(aCompoundCommand1);
+        aCompoundCommand4.add(aMoveCommand2);
+        aCompoundCommand4.add(aDeletNodeCommand);
+    }
+
+    @Test
+    public void testExecute_ContainsSimpleCommands() 
+    {
+    	assertFalse(aGraph.getRootNodes().contains(aClassNode));
+        assertFalse(aGraph.getEdges().contains(aCallEdge));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(0, 0, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+       
+        aCompoundCommand1.execute();        
+        
+        assertTrue(aGraph.getRootNodes().contains(aClassNode));
+        assertTrue(aGraph.getEdges().contains(aCallEdge));
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
+    }
+    
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testUndo_ContainsSimpleCommands() 
+    {
+        aCompoundCommand1.execute();
+        assertEquals(aClassNode.getBounds(), new Rectangle2D.Double(5, 5, aClassNode.getBounds().getWidth(), aClassNode.getBounds().getHeight()));
```

Comment:
you can make the test simple by removing lines 105-117. There's no reason the NodesToBeRemoved list should be true at this point.